api/v1beta1/nutanixcluster_types.go

```diff
@@ -46,14 +46,12 @@ type NutanixClusterSpec struct {
 
 	// ControlPlaneEndpoint represents the endpoint used to communicate with the control plane.
 	// host can be either DNS name or ip address
-	// +optional
 	ControlPlaneEndpoint capiv1.APIEndpoint `json:"controlPlaneEndpoint"`
 
 	// prismCentral holds the endpoint address and port to access the Nutanix Prism Central.
 	// When a cluster-wide proxy is installed, by default, this endpoint will be accessed via the proxy.
 	// Should you wish for communication with this endpoint not to be proxied, please add the endpoint to the
 	// proxy spec.noProxy list.
-	// +optional
 	PrismCentral *credentialTypes.NutanixPrismEndpoint `json:"prismCentral"`
```

Comment:
If PrismCentral is not present, currently the behavior moves to using prism central configuration from management cluster. Are we making a deliberate documented choice enforce prism central presence?

Reply:
to clarify, as long as we understand what the change will stop supporting, I'm all for explicitly specifying and not relying on defaults.

Reply:
for capx, to create cluster, prismCentral endpoint is a requirement from day 0 to be present in NutanixCluster. where do we document that we will take the endpoint details from mgmt? https://opendocs.nutanix.com/capx/v1.3.x/credential_management/

Reply:
the change will make sure that in API, we mention that in NutanixCluster, one has to pass PrismCentral and controlPlaneEndpoint always and its not optional (and was never optional)